src/python-bindings/classad2/classad.cpp

```diff
@@ -1 +1,713 @@
+        if( py_v_int == NULL ) {
+            // This was ClassAdInternalError in version 1.
+            PyErr_SetString( PyExc_RuntimeError, "Unknown ClassAd value type." );
+            return NULL;
+        }
+        long lv = PyLong_AsLong(py_v_int);
+        Py_DecRef(py_v_int);
+
+        if( lv == 1<<0 ) {
+            v.SetErrorValue();
+            return classad::Literal::MakeLiteral(v);
+        } else if( lv == 1<<1 ) {
+            v.SetUndefinedValue();
+            return classad::Literal::MakeLiteral(v);
+        } else {
+            // This was ClassAdInternalError in version 1.
+            PyErr_SetString( PyExc_RuntimeError, "Unknown ClassAd value type." );
+            return NULL;
+        }
+    }
+
+    if( py_is_datetime_datetime(py_v) ) {
+        // We do this song-and-dance in a bunch of places...
+        static PyObject * py_classad2_module = NULL;
+        if( py_classad2_module == NULL ) {
+             py_classad2_module = PyImport_ImportModule( "classad2" );
+        }
+
+        PyObject * py_ts = PyObject_CallMethod(
+            py_classad2_module,
+            "_convert_local_datetime_to_utc_ts",
+            "O", py_v
+        );
+        if( py_ts == NULL ) {
+            // PyObject_CallMethod() has already set an exception for us.
+            return NULL;
+        }
+
+        classad::abstime_t atime;
+        atime.offset = 0;
+        atime.secs = PyLong_AsLong(py_ts);
+        if( PyErr_Occurred() ) {
+            // PyLong_AsLong() has already set an exception for us.
+            return NULL;
+        }
+
+        v.SetAbsoluteTimeValue(atime);
+        return classad::Literal::MakeLiteral(v);
+    }
+
+    if( PyBool_Check(py_v) ) {
+        v.SetBooleanValue( py_v == Py_True );
+        return classad::Literal::MakeLiteral(v);
+    }
+
+    if( PyUnicode_Check(py_v) ) {
+        PyObject * py_bytes = PyUnicode_AsUTF8String(py_v);
+
+        char * buffer = NULL;
+        Py_ssize_t size = -1;
+        if( PyBytes_AsStringAndSize(py_bytes, &buffer, &size) == -1 ) {
+            // PyBytes_AsStringAndSize() has already set an exception for us.
+            return NULL;
+        }
+        v.SetStringValue( buffer, size );
+
+        Py_DecRef(py_bytes);
+        return classad::Literal::MakeLiteral(v);
+    }
+
+    if( PyBytes_Check(py_v) ) {
+        char * buffer = NULL;
+        Py_ssize_t size = -1;
+        if( PyBytes_AsStringAndSize(py_v, &buffer, &size) == -1 ) {
+            // PyBytes_AsStringAndSize() has already set an exception for us.
+            return NULL;
+        }
+        v.SetStringValue( buffer, size );
+        return classad::Literal::MakeLiteral(v);
+    }
+
+    if( PyLong_Check(py_v) ) {
+        // This should almost certainly be ...AsLongLong(), but the
+        // error is in the original version and in the classad3 module,
+        // this function won't exist anyway.
+        v.SetIntegerValue(PyLong_AsLong(py_v));
```

Comment:
I think we should change it to AsLongLong().

Reply:
I guess it's fair to assume that nobody is depending on integer truncation.